lib/eslint/eslint-helpers.js

```diff
@@ -800,6 +801,9 @@ function processOptions({
     if (fixTypes !== null && !isFixTypeArray(fixTypes)) {
         errors.push("'fixTypes' must be an array of any of \"directive\", \"problem\", \"suggestion\", and \"layout\".");
     }
+    if (!isEmptyArrayOrArrayOfNonEmptyString(flags)) {
+        errors.push("'flags' must be an array of non-empty strings.");
+    }
```

Comment:
Unknown flags would be silently ignored?

Reply:
Good question! I'm not sure what I would expect to happen when an unknown flag is encountered. Should it error or warn?

Reply:
Updated so unknown flags error out.